### PR TITLE
Rename Bio.SeqFeature.FeatureLocation to SimpleLocation

### PR DESCRIPTION
**Pull request by peterjc**
- [x] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [x] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit``
locally, and understand that continuous integration checks will be used to
confirm the Biopython unit tests and style checks pass with these changes.

- [x] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

This extracts the biggest change from #4063 (in terms of number of lines and files changed).

It is fairly simple to rebase the #4063 branch on top of this (I have done it locally and could update that pull request), leaving the ``.fromstring`` factory method and other position object work.

### CODE REVIEW

**Comment by mdehoon** [comment]
@peterjc  I have no objections, but how do I do that rebase thingy?

**Comment by peterjc** [comment]
I think GitHub would let me update your https://github.com/mdehoon/biopython/tree/seqfeature_cleanup branch (or anyone else with Biopython write permissions because there is a pull request using it - otherwise random people can't touch your code), so I'd do the rebase locally and then ``git push mdehoon seqfeature_cleanup --force`` to update your branch. If that needed further changes, then we'd need to update your local copy of the branch.

To do the rebase yourself: Once this is merged and you've done ``git fetch origin`` to grab the changes, you'd run ``git checkout seqfeature_cleanup`` (change to your branch), and ``git rebase origin/master``. Then the tricky part about dealing with merge conflicts - this isn't trivial enough to recommend as a learning exercise.

--

I'm curious if anyone else wants to comment on the class renaming...

File: Bio/Graphics/GenomeDiagram/_Feature.py [hunk 1]
```diff
@@ -129,7 +129,7 @@ def __process_feature(self):
         """Examine wrapped feature and set some properties accordingly (PRIVATE)."""
         self.locations = []
         bounds = []
-        # This will be a list of length one for simple FeatureLocation:
+        # This will be a list of length one for simple SimpleLocation:
```
Comment (mdehoon) [inline Bio/Graphics/GenomeDiagram/_Feature.py:132]: drop simple here

Comment (peterjc) [reply Bio/Graphics/GenomeDiagram/_Feature.py:132]: Yes, can do. That was done on your branch, doing it here too ought not to complicate the merge/rebase.

**Comment by mdehoon** [comment]
@peterjc  Time to merge?

**Comment by peterjc** [comment]
A shame we didn't get any additional comments, hopefully there will be no objections.

**Pull request merged**